src/game/objects/general/bridge.c

```diff
@@ -197,67 +215,73 @@ void Bridge_FlatFloor(
 void Bridge_FlatCeiling(
     ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
 {
-    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, y, z, item)) {
+    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, z, item)) {
         return;
     }
 
-    if (y > item->pos.y) {
-        *height = item->pos.y + STEP_L;
+    if (y <= item->pos.y || item->pos.y <= *height) {
+        return;
     }
+
+    *height = (int16_t)item->pos.y + STEP_L;
 }
 
 void Bridge_Tilt1Floor(
     ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
 {
-    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, y, z, item)) {
+    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, z, item)) {
         return;
     }
 
-    int32_t level = item->pos.y + (Bridge_GetOffset(item, x, y, z) >> 2);
-    if (y > level) {
+    int32_t offset_height = item->pos.y + (Bridge_GetOffset(item, x, y, z));
+    if (y > offset_height || item->pos.y >= *height) {
         return;
     }
 
-    *height = level;
+    *height = offset_height;
 }
 
 void Bridge_Tilt1Ceiling(
     ITEM_INFO *item, int32_t x, int32_t y, int32_t z, int16_t *height)
 {
-    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, y, z, item)) {
+    if (g_Config.fix_bridge_collision && !Bridge_IsSameSector(x, z, item)) {
         return;
     }
 
-    int32_t level = item->pos.y + (Bridge_GetOffset(item, x, y, z) >> 2);
-    if (y > level) {
-        *height = level + STEP_L;
+    int32_t offset_height = item->pos.y + (Bridge_GetOffset(item, x, y, z));
+    if (y <= offset_height || item->pos.y <= *height) {
+        return;
     }
+
+    *height = (int16_t)offset_height + STEP_L;
```

Comment:
Unnecessary cast

Reply:
`height` is `int16_t` while `offset_height` is `int32_t`. Should this have a cast?